pkg/odo/cli/list/namespace/namespace.go

```diff
@@ -57,18 +58,51 @@ func (plo *NamespaceListOptions) Validate(ctx context.Context) (err error) {
 }
 
 // Run contains the logic for the odo list project command
-func (plo *NamespaceListOptions) Run(ctx context.Context) error {
-	namespaces, err := plo.clientset.ProjectClient.List()
+func (plo *NamespaceListOptions) Run(_ context.Context) error {
+	namespaces, err := plo.run()
 	if err != nil {
 		return err
 	}
 
 	return HumanReadableOutput(os.Stdout, namespaces, plo.commandName)
 }
 
+func (plo *NamespaceListOptions) run() (api.ResourcesList, error) {
+	namespaces, err := plo.clientset.ProjectClient.List()
+	if err != nil {
+		return api.ResourcesList{}, err
+	}
+
+	if len(namespaces.Items) == 0 {
+		return api.ResourcesList{}, fmt.Errorf("you are not a member of any %[1]ss. You can request a %[1]s to be created using the `odo create %[1]s <%[1]s_name>` command", plo.commandName)
```

Comment:
Just noticed that `<` and `>` characters are escaped when marshaling the JSON output.

```
$ odo list namespaces -o json
{
        "message": "you are not a member of any namespaces. You can request a namespace to be created using the `odo create namespace \u003cnamespace_name\u003e` command"
}
```

From [this post on SO](https://stackoverflow.com/questions/28595664/how-to-stop-json-marshal-from-escaping-and), it seems this is the behavior of `json.Marshal`, but might be customized using a custom encoder.
I'd suggest removing those two characters here:

```go
return api.ResourcesList{}, fmt.Errorf("you are not a member of any %[1]ss. You can request a %[1]s to be created using the 'odo create %[1]s $%[1]s' command", plo.commandName)
```

What do you think?

Reply:
> What do you think?

This comment made me look at other `odo list` commands' JSON outputs more consciously, and below is what I see:
```sh
$ odo list components -o json
{}

$ odo list bindings -o json  
{}

$ odo list services -o json
{}

$ odo list namespaces -o json
{
	"message": "you are not a member of any namespaces. You can request a namespace to be created using the `odo create namespace \u003cnamespace_name\u003e` command"
}

```
I'd prefer consistency in JSON outputs, and hence shows just a `{}` as output if the user doesn't have access to any namespaces on the cluster.

WDYT?

Reply:
PTAL 61b8479.